File: example/learning-agent/java/src/main/java/org/aiddl/example/learning_agent/functions/PlanSuccessChecker.java
package org.aiddl.example.learning_agent.functions;

import java.util.Random;

import org.aiddl.common.java.planning.PlanningTerm;
import org.aiddl.core.java.interfaces.Function;
import org.aiddl.core.java.representation.SetTerm;
import org.aiddl.core.java.representation.Term;
import org.aiddl.core.java.tools.Logger;

public class PlanSuccessChecker implements Function {

	Random r;
	int next_action = 0;
		
	@Override
	public Term apply( Term args ) {
		SetTerm goal = (SetTerm) args.get(PlanningTerm.Goal);
		SetTerm state = (SetTerm) args.get(PlanningTerm.State);
		
		boolean success = state.containsAll(goal);
		
		if ( success ) {
			Logger.msg(this.getClass().getSimpleName(), ">>>>>>>>>>>>>>>>>>>>>>>>>>>");
			Logger.msg(this.getClass().getSimpleName(), ">>> GOAL REACHED BY PLAN!");
			Logger.msg(this.getClass().getSimpleName(), ">>>>>>>>>>>>>>>>>>>>>>>>>>>");
			try {
				Thread.sleep(1000);
			} catch (InterruptedException e1) {
				e1.printStackTrace();
			}
		} else {
			Logger.msg(this.getClass().getSimpleName(), ">>>>>>>>>>>>>>>>>>>>>>>>>>>");
			Logger.msg(this.getClass().getSimpleName(), ">>> GOAL FAILED...");
			Logger.msg(this.getClass().getSimpleName(), ">>>>>>>>>>>>>>>>>>>>>>>>>>>");
		}
		return Term.bool(success);
	}	
}

File: example/learning-agent-scala/src/main/scala/org/aiddl/example/learning_agent_scala/function/PlanSuccessChecker.scala
package org.aiddl.example.learning_agent_scala.function

import org.aiddl.core.scala.function.{Function, Verbose}
import org.aiddl.core.scala.representation.{CollectionTerm, KeyVal, ListTerm, SetTerm, Sym, Term, Tuple, Bool}
import org.aiddl.common.scala.planning.PlanningTerm

object PlanSuccessChecker extends Function with Verbose {

  override def apply(x: Term): Term = {
    val goal = x(PlanningTerm.Goal).asSet
    val state = x(PlanningTerm.State).asSet
    Bool(this(goal, state))
  }

  def apply(state: SetTerm, goal: SetTerm): Boolean = {
    val success = state.containsAll(goal)

    if (success) {
      this.logger.info(">>>>>>>>>>>>>>>>>>>>>>>>>>>")
      this.logger.info(">>> GOAL REACHED BY PLAN!")
      this.logger.info(">>>>>>>>>>>>>>>>>>>>>>>>>>>")
      try {
        Thread.sleep(1000)
      } catch {
        case e1: InterruptedException =>
          e1.printStackTrace()
      }
    } else {
      this.logger.info(">>>>>>>>>>>>>>>>>>>>>>>>>>>")
      this.logger.info(">>> GOAL FAILED...")
      this.logger.info(">>>>>>>>>>>>>>>>>>>>>>>>>>>")
    }
    success
  }
}
